tests/integration/new_relations/test_relations_coherence.py

```diff
@@ -0,0 +1,157 @@
+            num_units=1,
+            series=CHARM_SERIES,
+            config={"profile": "testing"},
+        )
+        await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=3000)
+
+        # Creating first time relation with user role
+        await ops_test.model.deploy(DATA_INTEGRATOR_APP_NAME)
+        await ops_test.model.applications[DATA_INTEGRATOR_APP_NAME].set_config({
+            "database-name": DATA_INTEGRATOR_APP_NAME.replace("-", "_"),
+        })
+        await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR_APP_NAME], status="blocked")
+        await ops_test.model.add_relation(DATA_INTEGRATOR_APP_NAME, DATABASE_APP_NAME)
+        await ops_test.model.wait_for_idle(apps=APP_NAMES, status="active")
+
+        connection_string = await build_connection_string(
+            ops_test,
+            DATA_INTEGRATOR_APP_NAME,
+            "postgresql",
+            database=DATA_INTEGRATOR_APP_NAME.replace("-", "_"),
+        )
+
+        connection = psycopg2.connect(connection_string)
+        connection.autocommit = True
+        cursor = connection.cursor()
+        try:
+            random_name = (
+                f"test_{''.join(secrets.choice(string.ascii_lowercase) for _ in range(10))}"
+            )
+            cursor.execute(f"CREATE DATABASE {random_name};")
+            assert False, "user role was able to create database"
+        except psycopg2.errors.InsufficientPrivilege:
+            pass
+        finally:
+            connection.close()
+
+        with psycopg2.connect(connection_string) as connection:
+            connection.autocommit = True
+            with connection.cursor() as cursor:
+                # Check that it's possible to write and read data from the database that
+                # was created for the application.
+                cursor.execute("DROP TABLE IF EXISTS test;")
+                cursor.execute("CREATE TABLE test(data TEXT);")
+                cursor.execute("INSERT INTO test(data) VALUES('some data');")
+                cursor.execute("SELECT data FROM test;")
+                data = cursor.fetchone()
+                assert data[0] == "some data"
+        connection.close()
+
+        await ops_test.model.applications[DATABASE_APP_NAME].remove_relation(
+            f"{DATABASE_APP_NAME}:database", f"{DATA_INTEGRATOR_APP_NAME}:postgresql"
+        )
+
+        # Re-relation with admin role with checking permission
+        await ops_test.model.applications[DATA_INTEGRATOR_APP_NAME].set_config({
+            "database-name": DATA_INTEGRATOR_APP_NAME.replace("-", "_"),
+            "extra-user-roles": "admin",
+        })
+        await ops_test.model.wait_for_idle(apps=[DATA_INTEGRATOR_APP_NAME], status="blocked")
+        await ops_test.model.add_relation(DATA_INTEGRATOR_APP_NAME, DATABASE_APP_NAME)
+        await ops_test.model.wait_for_idle(apps=APP_NAMES, status="active")
+
+        connection_string = await build_connection_string(
```

Comment:
@marceloneppel please consider to assign https://warthogs.atlassian.net/browse/DPE-2278 to Lucas the next pulse. DPE lib supports it for Mongo already, we should expand it for PostgreSQL (and MySQL?) to simplify our life here and users copy&paste life as well :-)